File: original_matlab_code/plotMapAllocation_stage2.m
function plotMapAllocation_stage2(X, na, color, name)

    for i = 1:na      
%         xx = reshape(X(:,i,:),[4,n_rounds+1]);
        xx=X{i};
%         xx = reshape(X(:,i,:),[4,size(X,3)]);
        if ~isempty(xx)
            plot(xx(1,:),xx(2,:), ':', 'Color', color(i,:), 'LineWidth', 2, 'DisplayName', name);
        end
    end

end

File: test/plotMapAllocation_stage2.py
import matplotlib.pyplot as plt


def plotMapAllocation_stage2(X, na, color, name):
    for i in range(na):
        xx = X[i]
        if xx is not None:
            plt.plot(xx[0, :], xx[1, :], ":", color=color[i], linewidth=2, label=name)
